src/main/groovy/de/gliderpilot/gradle/semanticrelease/SemanticReleaseChangeLogService.groovy

```diff
@@ -237,6 +242,7 @@ class SemanticReleaseChangeLogService {
 
         Release release = repo.releases().create(tag)
         new Release.Smart(release).body(changeLog(commits(Version.valueOf(version.previousVersion)), version).toString())
+        releaseAsset(release.assets(), "$version.version")
```

Comment:
You are right. URLConnection.guessContentTypeFromName() doesn't know a lot of file types. E.g. interestingly it does not support jar files.

Do you know, what github uses this information for? I currently have `application/octet-stream` as fallback. 

The possibility to rename an asset or give a description -- which does not seem to be possible with jcabi-github, but is part of the github-api (https://developer.github.com/v3/repos/releases/#upload-a-release-asset) -- is a valid one. With my changes, the only way to achive giving a different name would be to create a copy-task and use that copy instead of the original file. 

Thinking about this: by calling project.files() inside the plugin code masks the fact, that this only works for tasks that produce a single file (like Jar and Zip), but not for other tasks (like Copy. By removing this code in the ChangeLogService again and changing the method to `void releaseAssets(FileCollection)` the code of ChangeLogService would become more clear again, but the user of the plugin would have to write a little more code. 

Or there could be some overloaded methods `Asset releaseAsset(AbstractArchiveTask)`, `Asset releaseAsset(File)` and a new Class `Asset` with methods `Asset contentType(String)`, `Asset name(String)` and `Asset label(String)`. This would allow to write

```groovy
semanticRelease {
    changeLog {
        releaseAsset jar contentType 'application/zip' name 'theapplication.jar' label 'The application'
        // or
        releaseAsset(sourcesJar).contentType("application/zip").name("thesources.jar").label("the sources jar")
    }
}
```

By changing the methods in the ChangeLogService to `Asset releaseAsset(Map params = [:], AbstractArchiveTask task)`, the user could write `releaseAsset sourcesJar, contentType: 'application/zip', name: 'thesources.jar', label: 'the sources jar'` in their build file.